src/cmd/cli/command/mcp.go

```diff
@@ -0,0 +1,378 @@
+				}
+
+				// Add or update the Defang MCP server config
+				serversMap["defang"] = getVSCodeServerConfig()
+
+				mcpMap["servers"] = serversMap
+			}
+
+			existingData["mcp"] = mcpMap
+		}
+	} else {
+		// File doesn't exist, create a new config with minimal settings
+		existingData = map[string]interface{}{
+			"mcp": map[string]interface{}{
+				"servers": map[string]interface{}{
+					"defang": getVSCodeServerConfig(),
+				},
+			},
+		}
+	}
+
+	// Write the config to the file
+	data, err := json.MarshalIndent(existingData, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal config: %w", err)
+	}
+
+	if err := os.WriteFile(configPath, data, 0644); err != nil {
+		return fmt.Errorf("failed to write config file: %w", err)
+	}
+
+	return nil
+}
+
+var mcpSetupCmd = &cobra.Command{
+	Use:         "setup",
+	Short:       "Setup MCP client for defang mcp server",
+	Annotations: authNeededAnnotation,
+	Args:        cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, _ := cmd.Flags().GetString("client")
+		// Validate client
+		if !IsValidClient(client) {
+			return fmt.Errorf("invalid MCP client: %s. Valid MCP clients are: %s", client, GetValidClientsString())
+		}
+
+		fmt.Printf("Setting up MCP client for: %s\n", client)
+
+		// Get the config path for the client
+		configPath, err := getClientConfigPath(client)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("Config path: %s\n", configPath)
+
+		// Create the directory if it doesn't exist
+		configDir := filepath.Dir(configPath)
+		if err := os.MkdirAll(configDir, 0755); err != nil {
+			return fmt.Errorf("failed to create config directory: %w", err)
+		}
+
+		// Handle VSCode settings.json specially
+		if client == "vscode" {
+			if err := handleVSCodeConfig(configPath); err != nil {
+				return err
+			}
+		} else {
+			// For all other clients, use the standard format
+			var config MCPConfig
+
+			// Check if the file exists
+			if _, err := os.Stat(configPath); err == nil {
+				// File exists, read it
+				data, err := os.ReadFile(configPath)
+				if err != nil {
+					return fmt.Errorf("failed to read config file: %w", err)
+				}
+
+				// Parse the JSON
+				if err := json.Unmarshal(data, &config); err != nil {
+					// If we can't parse it, start fresh
+					config = MCPConfig{
+						MCPServers: make(map[string]MCPServerConfig),
+					}
+				}
+			} else {
+				// File doesn't exist, create a new config
+				config = MCPConfig{
+					MCPServers: make(map[string]MCPServerConfig),
+				}
+			}
+
+			// Add or update the Defang MCP server config
+			config.MCPServers["defang"] = getDefangMCPConfig()
+
+			// Write the config to the file
+			data, err := json.MarshalIndent(config, "", "  ")
+			if err != nil {
+				return fmt.Errorf("failed to marshal config: %w", err)
+			}
+
+			if err := os.WriteFile(configPath, data, 0644); err != nil {
+				return fmt.Errorf("failed to write config file: %w", err)
+			}
+		}
```

Comment:
Let's move all of this into a separate function and then move that function and the supporting functions and types into a new file (`src/pkg/mcp/setup.go` or something).